routes/rss.js

```diff
@@ -23,12 +25,36 @@ async function combinedRSSFeedRoute(req, res) {
 
           const author = blogRegistry[url]['chosenName'];
           const blogUrl = url;
-          const publishedAt = new Date(feed.items[0].pubDate);
 
-          return feed.items.map((item) => ({
-            item,
-            metadata: { author, blogUrl, publishedAt },
-          }));
+          return feed.items.map((item) => {
+            let blogPostUrl = item.link || '';
+            const publishedAt = item.pubDate
+              ? new Date(item.pubDate)
+              : new Date();
+
+            // Sometimes, it seems that `item.link` is not a full URL, so we
+            // need to join it with the blog URL.
+            if (!blogPostUrl.startsWith('http')) {
+              blogPostUrl = path.join(blogUrl, blogPostUrl);
+            }
+
+            const stringToHash = `${blogPostUrl}::${publishedAt.toISOString()}`;
+            const guid = crypto
```

Comment:
what's this for?

Reply:
This creates a stable guid for the blog posts so that RSS readers can remember that you read it. Previously, I used the id that comes from the blog hosts, but they can inconsistent and sometimes even non-unique. This way they are stable.

Reply:
awesome